templates/an-cash-dao/contracts/ANCashDAOTemplate.sol

```diff
@@ -0,0 +1,164 @@
+/*
+ * SPDX-License-Identifier:    MIT
+ */
+
+pragma solidity 0.4.24;
+
+import "./TokenCache.sol";
+import "./BaseTemplate.sol";
+
+import "./lib/os/ERC20.sol";
+
+
+contract ANCashDAOTemplate is BaseTemplate, TokenCache {
+    string constant private ERROR_MISSING_CACHE = "TEMPLATE_MISSING_TOKEN_CACHE";
+    string constant private ERROR_EMPTY_HOLDERS = "COMPANY_EMPTY_HOLDERS";
+    string constant private ERROR_BAD_HOLDERS_STAKES_LEN = "COMPANY_BAD_HOLDERS_STAKES_LEN";
+    string constant private ERROR_BAD_VOTE_SETTINGS = "COMPANY_BAD_VOTE_SETTINGS";
+    string constant private ERROR_BAD_COLLATERAL_REQUIREMENT_SETTINGS = "COMPANY_BAD_COL_REQ_SETTINGS";
+
+    bool constant private SET_APP_FEES_CASHIER = false;
+    bool constant private TOKEN_TRANSFERABLE = true;
+    uint8 constant private TOKEN_DECIMALS = uint8(18);
+    uint256 constant private TOKEN_MAX_PER_ACCOUNT = uint256(0);
+    uint64 constant private DEFAULT_FINANCE_PERIOD = uint64(30 days);
+
+    struct Cache {
+        address dao;
+        address token;
+        address tokenManager;
+        address agreement;
+    }
+
+    mapping (address => Cache) internal cache;
+
+    constructor(DAOFactory _daoFactory, ENS _ens, MiniMeTokenFactory _miniMeFactory, IFIFSResolvingRegistrar _aragonID)
+        BaseTemplate(_daoFactory, _ens, _miniMeFactory, _aragonID)
+        public
+    {
+        _ensureAragonIdIsValid(_aragonID);
+        _ensureMiniMeFactoryIsValid(_miniMeFactory);
+    }
+
+    function setupDAO(string _id, string _name, string _symbol, address[] _holders, uint256[] _stakes) external {
```

Comment:
Rather than name this `setupDAO()`, let's prefer naming it `createDAO()` or `initializeDAO()`.

The `SetupDAO` event is our standard event in templates to mark that the template is finished creating the org and can be confusing since `setupDAO()` is the first step here.